### ISSUE TITLE
Workaround for flickering output with start/stop (or toggle) in continuous Terminal use case 

### ISSUE THREAD

**Issue by aral** [mattboldt/typed.js#420]
## To reproduce

1. Create an array of strings where each string builds upon the last (e.g., to create a Terminal effect.) 

2. After each string typed event, stop Typed.js.

3. Create a toggle button that stops/starts Typed.js. Have the toggle button disabled on start and enabled on stop to ensure it is not activated mid-typing animation.

4. Quickly tap the toggle button after Typed.js has stopped (within 700ms).

## What should happen

The animation should start from the next step.

## What actually happens

The animation starts but flickers and jumps to the string after the current string.

## Why it happens

This is due to the `backspace` timeout being called, by default, 700ms after Typed.js stops. It manifests as a bug in my use case but I’m not sure this is a bug per se. It is definitely surprising behaviour.

## Workaround

Set `backDelay` to `0` when initialising your animation.

**Comment by EMMANATT** [comment]
The toggle button disabled on start and enabled , the next step.

**Comment by mattboldt** [comment]
Thanks for such a detailed issue! Yeah it does look like the `backDelay` timeout is a root cause. I'm looking into a fix.

**Comment by strygo** [comment]
There might be a cleaner solution to this race condition, but I was able to address it by saving the pause status in keepTyping() and doneTyping() and then clearing the timeout when invoking stop(). This ensures that start() can pick up where it needs to and no timeout happens after stop has occurred.

I encountered the problem in a slightly different scenario: when pausing and resuming the typing based on whether the text was in the viewport or not. If the timeout associated with backDelay fired after leaving and re-entering the view, the typing and backing would fight each other.
